specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json

```diff
@@ -1075,6 +1107,39 @@
           }
         }
       },
+      "patch": {
+        "operationId": "ContinuousDataExports_Update",
+        "summary": "Update an existing continuous data export via PATCH.",
+        "x-ms-examples": {
+          "Update export": {
+            "$ref": "./examples/continuousdataexports_update.json"
+          }
+        },
+        "parameters": [
+          {
+            "in": "path",
+            "name": "export_id",
+            "description": "Unique ID for the continuous data export.",
+            "type": "string",
+            "required": true
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Data export patch body.",
+            "schema": {},
```

Comment:
Why is this empty?  Comment applies to all `PATCH` operations being added.

Reply:
Thanks, I've made them all objects in the latest commit

Reply:
Why objects though?  The `PUT` operation for this specific `PATCH` is a `#/definitions/ContinuousDataExport`.

Reply:
Doesn't that imply that the PATCH body must include ALL required fields in `#/definitions/ContinuousDataExport`? If that's not the assumption, I'll make the schema match with the PUT in each case

Reply:
It looks like `ContinuousDataExport` contains read-only content so specifying it probably isn't appropriate.  Usually it's best to create types like `ContinuousDataExport` as a combination of types, where the outer one (i.e. the thing returned) contains the read-only content and the inner "properties" contains user-modifiable content.  You can see an example of this in [RedisResource](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/redis/resource-manager/Microsoft.Cache/stable/2020-06-01/redis.json#L2200), where the inner type `RedisProperties` contains user-modifiable content.

CC @johanste

Reply:
Leaving it as simply `object` means callers don't get a strongly-typed parameter which isn't a good user experience.

Reply:
I will defer this part to @clecompt-msft in January. Changing all of the existing types to separate out the read-only vs the optional will need to be considered later. In the mean time, can we get away with the schema being empty or as object? I'm following the Azure Digital Twins stable API as a guideline right now https://github.com/Azure/azure-rest-api-specs/blob/c05f4de5e6f52b8b1941f39fe22094d264b75d9b/specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2020-10-31/digitaltwins.json#L200

Reply:
How do callers know what object to provide in this case?

Reply:
IIRC digital twins specifies an object because their schema is free-flowing.  If you have a stable schema what the API accepts then you should properly model it.

Reply:
You can't express JSON-merge-patch types correctly in Swagger v2. OR-types and the "nullable" keyword weren't introduced until v3. We decided it was better to opt out of explicit typing rather than provide wrong types (it also prevents us from having to do a bunch of nasty internal hacks to work around that discrepancy).

Reply:
@jhendrixMSFT @lirenhe Can you help us out with this?